migrations/1730965858452-update-resources-file-storage.ts

```diff
@@ -0,0 +1,315 @@
+                },
+              },
+              as: 'fileField',
+              in: '$$fileField.name',
+            },
+          },
+        },
+      },
+    ])
+  ).map((resource) => ({
+    resourceId: resource._id,
+    resourceName: resource.name,
+    fieldNames: resource.fieldNames,
+  }));
+
+  let totalFiles = 0;
+  let migratedFiles = 0;
+  try {
+    const driveId = await getDriveId();
+    // For each resource, get all records for that resource id and that in their data property, at least the file type field exists
+    await new Promise((resolve) => {
+      resources.forEach(
+        async ({ resourceId, resourceName, fieldNames }, indexResource) => {
+          const query = {
+            resource: resourceId,
+            $or: fieldNames.map((fieldName) => ({
+              [`data.${fieldName}`]: { $ne: null }, // Field exists and is not null
+            })),
+          };
+          const records = await Record.find(query);
+          const numberOfRecords = records.length;
+          for (const [index, r] of records.entries()) {
+            let filesToUpload = 0;
+            for (
+              let indexField = 0;
+              indexField < fieldNames.length;
+              indexField++
+            ) {
+              const field = fieldNames[indexField];
+              /** Check which of the file fields exists in the given resource */
+              if (r.data[field]) {
+                if (Array.isArray(r.data[field])) {
+                  const newFileField = [];
+                  filesToUpload = r.data[field].length;
+                  for (
+                    let indexFile = 0;
+                    indexFile < r.data[field].length;
+                    indexFile++
+                  ) {
+                    let file = r.data[field][indexFile];
+                    const fileData = await handleFile(file);
+                    if (!isNil(fileData)) {
+                      const { itemId } = await uploadFile(
+                        fileData,
+                        file.name,
+                        driveId
+                      );
+                      file = {
+                        ...file,
+                        content: { itemId, driveId },
+                      };
+                      logger.info(
+                        `Uploaded file ${
+                          indexFile + 1
+                        } of ${filesToUpload} for the record ${
+                          index + 1
+                        } of ${numberOfRecords} in ${resourceName}`
+                      );
+                      migratedFiles++;
+                    }
+                    newFileField.push(file);
+                  }
+                  r.data[field] = newFileField;
+                } else {
+                  filesToUpload = 1;
+                  const fileData = await handleFile(r.data[field]);
+                  if (!isNil(fileData)) {
+                    // const { itemId } = await uploadFile(
```

Comment:
@AntoineRelief  I forgot to uncomment this, but should be uncommented, same when file is array 👍